x-pack/legacy/plugins/infra/public/components/source_configuration/view_source_configuration_button.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton } from '@elastic/eui';
+import React from 'react';
+import { Route } from 'react-router-dom';
+
+export enum ViewSourceConfigurationButtonHrefBase {
+  infrastructure = 'infrastructure',
+  logs = 'logs',
+}
+
+interface ViewSourceConfigurationButtonProps {
+  'data-test-subj'?: string;
+  hrefBase: ViewSourceConfigurationButtonHrefBase;
+  children: React.ReactNode;
+}
+
+export const ViewSourceConfigurationButton = ({
+  'data-test-subj': dataTestSubj,
+  hrefBase,
+  children,
+}: ViewSourceConfigurationButtonProps) => {
+  const href = `/${hrefBase}/settings`;
+
+  return (
+    <Route
+      key={href}
+      path={href}
+      children={({ match, history }) => (
+        <EuiButton data-test-subj={dataTestSubj} color="primary" onClick={() => history.push(href)}>
```

Comment:
Maybe `useMemo` this `onClick` with the value of `href`?

Reply:
To use `useMemo` or `useCallback` here would mean moving that definition to the top of the component function scope, and there it's difficult to access `history`. I don't think in this instance there's much (if anything) to be gained by memoizing the handler - referential equality aside inline functions are rarely a bottleneck, and this component won't be re-rendering a lot (if at all).